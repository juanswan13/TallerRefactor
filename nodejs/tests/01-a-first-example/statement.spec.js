const statement = require('../../src/01-a-first-example/statement');

test('create a statemen', () => {
  const invoce = {
    customer: 'BigCo',
    performances: [
      {
        playID: 'hamlet',
        audience: 55
      },
      {
        playID: 'as-like',
        audience: 35
      },
      {
        playID: 'othello',
        audience: 40
      }
    ]
  };

  const plays = {
    hamlet: { name: 'Hamlet', type: 'tragedy' },
    'as-like': { name: 'As You Like It', type: 'comedy' },
    othello: { name: 'Othello', type: 'tragedy' }
  };

  expect(statement(invoce, plays)).toMatchSnapshot();
});

test('type does not exists', () => {
  const plays = {
    avenger: {
      name: 'Avenger',
      type: 'science-fiction'
    }
  };

  const invoces = {
    customer: 'New Enterprise',
    performances: [{
      playID: 'avenger',
      audience: 300
    }]
  };

  expect(() => statement(invoces, plays)).toThrowErrorMatchingSnapshot();
});

test('performances with little audience', () => {
  const plays = {
    hamlet: { name: 'Hamlet', type: 'tragedy' },
    'as-like': { name: 'As You Like It', type: 'comedy' },
    othello: { name: 'Othello', type: 'tragedy' }
  };

  const invoices = {
    customer: 'Alternative Performance',
    performances: [
      {
        playID: 'as-like',
        audience: 15
      },
      {
        playID: 'othello',
        audience: 25
      }
    ]
  };

  expect(statement(invoices, plays)).toMatchSnapshot();
});

test('performances with minimal audience', () => {
  const plays = {
    hamlet: { name: 'Hamlet', type: 'tragedy' },
    'as-like': { name: 'As You Like It', type: 'comedy' },
    othello: { name: 'Othello', type: 'tragedy' }
  };

  const invoices = {
    customer: 'Alternative Performance',
    performances: [
      {
        playID: 'as-like',
        audience: 20
      },
      {
        playID: 'othello',
        audience: 30
      }]
  };

  expect(statement(invoices, plays)).toMatchSnapshot();
});
