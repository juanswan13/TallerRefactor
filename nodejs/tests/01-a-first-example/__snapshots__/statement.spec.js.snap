// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create a statemen 1`] = `
"Statement for BigCo
  Hamlet: $650.00 (55 seats)
  As You Like It: $580.00 (35 seats)
  Othello: $500.00 (40 seats)
Amount owed is $1,730.00
You earned 47 credits
"
`;

exports[`performances with little audience 1`] = `
"Statement for Alternative Performance
  As You Like It: $345.00 (15 seats)
  Othello: $400.00 (25 seats)
Amount owed is $745.00
You earned 3 credits
"
`;

exports[`performances with minimal audience 1`] = `
"Statement for Alternative Performance
  As You Like It: $360.00 (20 seats)
  Othello: $400.00 (30 seats)
Amount owed is $760.00
You earned 4 credits
"
`;

exports[`type does not exists 1`] = `"unknown type: science-fiction"`;
